### PR TITLE
Fixing dynamic test codeRef generation for JUnit 5 v5.6

### PR DESCRIPTION
**Pull request by HardNorth**

### CODE REVIEW

File: src/main/java/com/epam/reportportal/junit5/ReportPortalExtension.java [hunk 1]
```diff
@@ -327,30 +338,24 @@ private void startTestItem(ExtensionContext context, List<Object> arguments, Str
...
 		return new TestCaseIdEntry(codeRef, codeRef.hashCode());
 	}
 
-	@Nullable
 	private TestCaseIdEntry getTestCaseId(Method method, String codeRef, List<Object> arguments) {
-		return ofNullable(method.getAnnotation(TestCaseId.class)).map(testCaseId -> {
-			if (testCaseId.parametrized()) {
-				return TestCaseIdUtils.getParameterizedTestCaseId(method, arguments);
+		if (method != null) {
+			TestCaseId tcId = method.getAnnotation(TestCaseId.class);
```
Comment (Yumfriez) [inline src/main/java/com/epam/reportportal/junit5/ReportPortalExtension.java]: Probably, `testCaseId` or `caseId` looks better

File: src/main/java/com/epam/reportportal/junit5/ReportPortalExtension.java [hunk 2]
```diff
@@ -327,30 +338,24 @@ private void startTestItem(ExtensionContext context, List<Object> arguments, Str
...
 	private TestCaseIdEntry getTestCaseId(String codeRef) {
 		return new TestCaseIdEntry(codeRef, codeRef.hashCode());
 	}
 
-	@Nullable
 	private TestCaseIdEntry getTestCaseId(Method method, String codeRef, List<Object> arguments) {
-		return ofNullable(method.getAnnotation(TestCaseId.class)).map(testCaseId -> {
-			if (testCaseId.parametrized()) {
-				return TestCaseIdUtils.getParameterizedTestCaseId(method, arguments);
+		if (method != null) {
```
Comment (Yumfriez) [inline src/main/java/com/epam/reportportal/junit5/ReportPortalExtension.java]: This `if(method != null)` with `tcId != null` and `return` statement can be replaced by ofNullable().flatMap().orElseGet() construction

File: src/main/java/com/epam/reportportal/junit5/ReportPortalExtension.java [hunk 3]
```diff
@@ -266,6 +264,33 @@ private void startTestItem(ExtensionContext context, String type) {
 		startTestItem(context, Collections.emptyList(), type, null);
 	}
 
+	private String getCodeRef(Method method) {
+		return method.getDeclaringClass().getCanonicalName() + "." + method.getName();
+	}
+
+	private static String appendSuffixIfNotEmpty(String str, String suffix) {
```
Comment (Yumfriez) [inline src/main/java/com/epam/reportportal/junit5/ReportPortalExtension.java]: Is `static` a 'must have' here?

File: src/main/java/com/epam/reportportal/junit5/ReportPortalExtension.java [hunk 4]
```diff
@@ -266,6 +264,33 @@ private void startTestItem(ExtensionContext context, String type) {
...
+		return str + (suffix.isEmpty() ? "" : "$" + suffix);
+	}
+
+	private String getCodeRef(ExtensionContext context, String currentCodeRef) {
+		Optional<Method> testMethod = context.getTestMethod();
+		if (testMethod.isPresent()) {
```
Comment (Yumfriez) [inline src/main/java/com/epam/reportportal/junit5/ReportPortalExtension.java]: Probably can be replaced with ofNullable().map().orElseGet() here

File: src/main/java/com/epam/reportportal/junit5/ReportPortalExtension.java [hunk 5]
```diff
@@ -266,6 +264,33 @@ private void startTestItem(ExtensionContext context, String type) {
...
+			return appendSuffixIfNotEmpty(getCodeRef(testMethod.get()), currentCodeRef);
+		} else {
+			String newCodeRef = appendSuffixIfNotEmpty(context.getDisplayName(), currentCodeRef);
+			Optional<ExtensionContext> parentContext = context.getParent();
+			if (parentContext.isPresent()) {
```
Comment (Yumfriez) [inline src/main/java/com/epam/reportportal/junit5/ReportPortalExtension.java]: Probably can be replaced with ofNullable().map().orElseGet() here

File: src/main/java/com/epam/reportportal/junit5/ReportPortalExtension.java (hunk 1 above)
Comment (HardNorth) [reply src/main/java/com/epam/reportportal/junit5/ReportPortalExtension.java]: What the problem?

File: src/main/java/com/epam/reportportal/junit5/ReportPortalExtension.java (hunk 3 above)
Comment (HardNorth) [reply src/main/java/com/epam/reportportal/junit5/ReportPortalExtension.java]: What the problem?

File: src/main/java/com/epam/reportportal/junit5/ReportPortalExtension.java (hunk 2 above)
Comment (HardNorth) [reply src/main/java/com/epam/reportportal/junit5/ReportPortalExtension.java]: Can't agree here, I need to write ofNullable 2 times on `method` parameter and on annotation get, and for both cases I need the last line as a fallback.

File: src/main/java/com/epam/reportportal/junit5/ReportPortalExtension.java (hunk 1 above)
Comment (HardNorth) [reply src/main/java/com/epam/reportportal/junit5/ReportPortalExtension.java]: OK

File: src/main/java/com/epam/reportportal/junit5/ReportPortalExtension.java (hunk 4 above)
Comment (HardNorth) [reply src/main/java/com/epam/reportportal/junit5/ReportPortalExtension.java]: OK

File: src/main/java/com/epam/reportportal/junit5/ReportPortalExtension.java (hunk 5 above)
Comment (HardNorth) [reply src/main/java/com/epam/reportportal/junit5/ReportPortalExtension.java]: OK

**Pull request merged**